modules/02_welfare/utilitarian/equations.gms

```diff
@@ -21,21 +21,57 @@ q02_welfareGlob..
 *' summing over all time steps taking into account the pure time preference rate.
 *' Assuming an intertemporal elasticity of substitution of 1, it holds:
 ***---------------------------------------------------------------------------
-q02_welfare(regi)..
-    v02_welfare(regi) 
+q02_welfare(regi) ..
+  v02_welfare(regi) 
   =e=
     sum(ttot $(ttot.val ge 2005),
-        pm_welf(ttot) * pm_ts(ttot) * (1 / ( (1 + pm_prtp(regi))**(pm_ttot_val(ttot)-2005) ) )
-        *   (  (pm_pop(ttot,regi) 
-                *   (
-                        ((( (vm_cons(ttot,regi))/pm_pop(ttot,regi))**(1-1/pm_ies(regi))-1)/(1-1/pm_ies(regi)) )$(pm_ies(regi) ne 1)
-                       + (log((vm_cons(ttot,regi)) / pm_pop(ttot,regi)))$(pm_ies(regi) eq 1)
-                    )
+      pm_welf(ttot)
+    * pm_ts(ttot)
+    * (1 / ((1 + pm_prtp(regi)) ** (pm_ttot_val(ttot) - 2005)))
+    * ( ( pm_pop(ttot,regi) 
+        * (
+            ( ( ( vm_cons(ttot,regi)
+                / pm_pop(ttot,regi)
                 )
-$if %cm_INCONV_PENALTY% == "on"  - v02_inconvPen(ttot,regi) - v02_inconvPenCoalSolids(ttot,regi)
-$if "%cm_INCONV_PENALTY_FESwitch%" == "on"  - sum((entySe,entyFe,te,sector,emiMkt)$(se2fe(entySe,entyFe,te) AND entyFe2Sector(entyFe,sector) AND sector2emiMkt(sector,emiMkt) AND (entySeBio(entySe) OR entySeSyn(entySe) OR entySeFos(entySe)) ), v02_NegInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt) + v02_PosInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt))/1e3	
-            )
+             ** (1 - (1 / pm_ies(regi)))
+              - 1
+              )
+            / (1 - (1 / pm_ies(regi)))
+            )$( pm_ies(regi) ne 1 )
+          + log((vm_cons(ttot,regi)) / pm_pop(ttot,regi))$( pm_ies(regi) eq 1 )
+          )
         )
+$ifthen %cm_INCONV_PENALTY% == "on"
+      - v02_inconvPen(ttot,regi)
+      - v02_inconvPenCoalSolids(ttot,regi)
+$endif
+$ifthen "%cm_INCONV_PENALTY_FESwitch%" == "on"
+        !! inconvenience cost for fuel switching in FE between fossil,
+        !! biogenic, synthetic solids, liquids and gases across sectors and
+        !! emissions markets
+      - sum((entySe,entyFe,te,sector,emiMkt)$(    
+                                                se2fe(entySe,entyFe,te)
+                                            AND entyFe2Sector(entyFe,sector)
+                                            AND sector2emiMkt(sector,emiMkt)
+                                            AND (   entySeBio(entySe) 
+                                                 OR entySeSyn(entySe)
+                                                 OR entySeFos(entySe))       ),
```

Comment:
Do you anticipate some other SE type besides fossil, biogenic, and synthetic being introduced to REMIND?

Reply:
no, this is just a non-thought-through replica of the penalty for final energy in general (`v02_NegInconvPenFeBioSwitch`).